### PR TITLE
SUBMARINE-1109. Java k8s client version from 6 upgrade to 11

### PR DESCRIPTION
**Pull request by noidname01**
### What is this PR for?
The version of k8s client we use is almost out of date, and lacks many features, like CustomResources Management, Generic Api and more concise function.

### What type of PR is it?
[Improvement]

### Todos

None

### What is the Jira issue?

SUBMARINE-1109. Java k8s client version from 6 upgrade to 11

### How should this be tested?

None

### Screenshots (if appropriate)

None

### Questions:
* Do the license files need updating? No
* Are there breaking changes for older versions? No
* Does this need new documentation? No

### CODE REVIEW

File: submarine-server/server-submitter/submitter-k8s/src/main/java/org/apache/submarine/server/submitter/k8s/K8sSubmitter.java [hunk 1]
```diff
@@ -28,32 +28,32 @@
...
+import io.kubernetes.client.openapi.apis.CustomObjectsApi;
+import io.kubernetes.client.openapi.models.V1DeleteOptionsBuilder;
+import io.kubernetes.client.openapi.models.V1Deployment;
+import io.kubernetes.client.openapi.models.CoreV1Event;
+import io.kubernetes.client.openapi.models.CoreV1EventList;
+import io.kubernetes.client.openapi.models.V1ObjectMeta;
+import io.kubernetes.client.openapi.models.V1PersistentVolumeClaim;
+import io.kubernetes.client.openapi.models.V1Pod;
+import io.kubernetes.client.openapi.models.V1PodList;
+//import io.kubernetes.client.openapi.models.V1Service;
```
Comment (pingsutw) [inline submarine-server/server-submitter/submitter-k8s/src/main/java/org/apache/submarine/server/submitter/k8s/K8sSubmitter.java:52]: Nit: remove it.
```suggestion
```

**Review by pingsutw — APPROVED** [review]
Others LGTM, thanks.

**Review by KUAN-HSUN-LI — APPROVED** [review]
+1

File: submarine-server/server-submitter/submitter-k8s/src/main/java/org/apache/submarine/server/submitter/k8s/K8sSubmitter.java [hunk 2]
```diff
@@ -539,9 +537,9 @@ public void createServe(ServeSpec spec)
       api.createNamespacedCustomObject(istioVirtualService.getGroup(),
               istioVirtualService.getVersion(),
               "default",
-              istioVirtualService.getPlural(),
-              istioVirtualService,
-              "true");
```
Comment (KUAN-HSUN-LI) [inline submarine-server/server-submitter/submitter-k8s/src/main/java/org/apache/submarine/server/submitter/k8s/K8sSubmitter.java:544]: This is a mistake of replace istioVirtualService with seldonDeployment

File: submarine-server/server-submitter/submitter-k8s/src/main/java/org/apache/submarine/server/submitter/k8s/K8sSubmitter.java [hunk 3]
```diff
@@ -570,16 +568,8 @@ public void deleteServe(ServeSpec spec)
...
-              seldonDeployment.getMetadata().getName(),
-              new V1DeleteOptionsBuilder().withApiVersion(seldonDeployment.getApiVersion()).build(),
-              null, null, null);
-      api.deleteNamespacedCustomObject(istioVirtualService.getGroup(),
-              istioVirtualService.getVersion(),
-              "default",
-              istioVirtualService.getPlural(),
-              istioVirtualService.getMetadata().getName(),
-              new V1DeleteOptionsBuilder().withApiVersion(istioVirtualService.getApiVersion()).build(),
-              null, null, null);
```
Comment (KUAN-HSUN-LI) [inline submarine-server/server-submitter/submitter-k8s/src/main/java/org/apache/submarine/server/submitter/k8s/K8sSubmitter.java:582]: Don't remove the istioVirtualService

**Review by KUAN-HSUN-LI — CHANGES_REQUESTED** [review]
There were some mistakes in fixing the conflict

File: submarine-server/server-submitter/submitter-k8s/src/main/java/org/apache/submarine/server/submitter/k8s/K8sSubmitter.java (hunk 2 above)
Comment (noidname01) [reply submarine-server/server-submitter/submitter-k8s/src/main/java/org/apache/submarine/server/submitter/k8s/K8sSubmitter.java:544]: Oh, you're right. I'll fix it ASAP.

**Comment by KUAN-HSUN-LI** [comment]
@noidname01 Some tests are failed. It seems like the import error

**Comment by noidname01** [comment]
> @noidname01 Some tests are failed. It seems like the import error

Yeah I know, seems that I screw it when rebasing🥲

**Comment by noidname01** [comment]
@KUAN-HSUN-LI fixed!

**Review by KUAN-HSUN-LI — APPROVED** [review]

**Pull request closed without merging**